lib/chef/knife/solo_cook.rb

```diff
@@ -260,14 +260,15 @@ def write(content, dest)
         file.unlink
       end
 
-      def rsync(source_path, target_path, extra_opts = '--delete-after')
+      def rsync(source_path, target_path, extra_opts = ['--delete-after', '-zt'])
         if config[:ssh_gateway]
           ssh_command = "ssh -TA #{config[:ssh_gateway]} ssh -T -o StrictHostKeyChecking=no #{ssh_args}"
         else
           ssh_command = "ssh #{ssh_args}"
         end
 
-        cmd = ['rsync', '-rL', rsync_debug, rsync_permissions, %Q{--rsh=#{ssh_command}}, extra_opts]
+        cmd = ['rsync', '-rL', rsync_debug, rsync_permissions, %Q{--rsh=#{ssh_command}}]
+        cmd += extra_opts
```

Comment:
There right-hand sides are different (list vs scalar):
```
2.2.2 :001 > part1 = [1,2,3]
 => [1, 2, 3]
2.2.2 :002 > part2 = [3,4,5]
 => [3, 4, 5]
2.2.2 :003 > part1 += part2
 => [1, 2, 3, 3, 4, 5]
2.2.2 :004 > part1
 => [1, 2, 3, 3, 4, 5]
2.2.2 :005 > part1 = [1,2,3]
 => [1, 2, 3]
2.2.2 :006 > part1 << part2
 => [1, 2, 3, [3, 4, 5]]
```

Though point taken. On consistency. If it really looks like the `cmd.flatten.compact` is necessary I could `<<` the whole thing. Otherwise I'll switch everything to be `+=`.

Thanks!